### PR TITLE
quiz-page

### PR DESCRIPTION
**Pull request by MikhailBiltsevich**
I've implemented html-web page without interactivity. I'm going to read about state hook and general info about hooks and add it. I think hooks is necessary, because I didn't use `React.Component` class. Also I realized additional component, that displays question-data.

Here the screenshot of my today's work (17/02/2021)
Снимок экрана (34)

### CODE REVIEW

**Comment by MikhailBiltsevich** [comment]
I've add Redux (reduxToolkit, react-redux and @types/react-redux). I created store and one slice. Also for data I add several interfaces in file `types.ts`

**Comment by MikhailBiltsevich** [comment]
I've implemented adding, editing and deleting questions. I didn't realize delete answers yet. I think my code should refactor, but it now it works. You can check functionality

File: src/pages/quiz/QuizPage.tsx [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    <div className='container'>
+      <div className="row">
+        <div className="col-md-5">
+          <form>
+            <fieldset>
+              <legend>Quiz</legend>
+              <InputComponent value={title} label='Title' onChange={(event) => { dispatch(setTitle(event.target.value)); }} type='text' id='quiz-title-input'></InputComponent>
+            </fieldset>
+
+            <fieldset>
```
Comment (DieselWolf) [inline src/pages/quiz/QuizPage.tsx:159]: would be great to create a new component, like:  <Fieldset/>

**Review by DieselWolf — CHANGES_REQUESTED** [review]

File: src/pages/quiz/QuizPage.tsx [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+import InputComponent from "../../components/InputComponent/InputComponent";
+import SelectComponent from "../../components/SelectComponent/SelectComponent";
+import CollapseSectionComponent from "../../components/CollapseSectionComponent/CollapseSectionComponent";
+import { BaseSyntheticEvent, SyntheticEvent, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { addQuestion, removeQuestion, selectQuestions, selectTitle, setTitle, updateQuestion } from '../../app/slices/quizSlice';
+import { IAnswer, IQuestion } from "../../app/types";
+
+const QuizPage = () => {
+  const questionType = [
```
[the remark this answers is not in the document]
Comment (MikhailBiltsevich) [reply src/pages/quiz/QuizPage.tsx:12]: I've moved this const to `src/app/types.ts`

File: src/app/slices/quizSlice.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    }
+  }
+});
+
+export const { setTitle, addQuestion, removeQuestion, updateQuestion } = quizSlice.actions;
+
+export const selectTitle = (state: IState) => state.quiz.title;
+export const selectQuestions = (state: IState) => state.quiz.questions;
+
+export default quizSlice.reducer;
```
[the remark this answers is not in the document]
Comment (MikhailBiltsevich) [reply src/app/slices/quizSlice.ts:35]: I'll read the diffrence later

File: src/pages/quiz/QuizPage.tsx (hunk 1 above)
Comment (MikhailBiltsevich) [reply src/pages/quiz/QuizPage.tsx]: I should create component for what? I can't understand by selected rows

**Comment by MikhailBiltsevich** [comment]
I've created two container components and some refactor. Also I've moved `questionTypes` const to types.ts file

**Review by DieselWolf — APPROVED** [review]

**Pull request merged**